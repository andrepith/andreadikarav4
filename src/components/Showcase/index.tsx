import React from "react";
import Image from "next/image";
import bioInterface from "@/lib/Types";

const Showcase = ({ bio }: bioInterface) => {
  return (
    <section className="justify-content-start showcase" id="project">
      <div className="container">
        <h2 className="section-title text-center">Project.</h2>
        <div className="row">
          {bio.portofolio.map((item, key) => (
            <div className="col-lg-6 col-sm-12 mt-3 showcase-cards" key={key}>
              <a href={item.url} target="__blank" className="showcase-card">
                <Image
                  className="showcase-image image"
                  src={item.image}
                  alt={item.alt}
                  layout="fill"
                />
                <div className="image-overlay">
                  <div className="image-overlay-text">
                    <p className="project-company mb-0">{item.name}</p>
                    <p className="project-title">{item.type}</p>
                  </div>
                </div>
              </a>
            </div>
          ))}
        </div>
      </div>
    </section>
  );
};

Showcase.displayName = "Showcase";

export default Showcase;
