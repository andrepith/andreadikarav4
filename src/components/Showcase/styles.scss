.showcase {
  margin-top: 48px;
  scroll-margin-top: 86px;
}

@include for-tablet-portrait-down {
  .showcase-cards {
    padding-left: 0;
    padding-right: 0;
  }
}

.showcase-card {
  width: 100%;
  height: 425px;
  background-color: $color-white;
  cursor: pointer;
  border-radius: 8px;
  display: block;
  position: relative;
  transition: 0.4s ease all;

  &:hover {
    filter: blur(0);
    opacity: 1;
  }

  @include for-tablet-portrait-down {
    border-radius: 0;

    &:hover {
      transform: none;
    }
  }
}

.showcase-image {
  position: relative !important;
  overflow: hidden;
  max-height: 85% !important;
  min-height: unset !important;
}

.image {
  min-width: unset !important;
  max-width: 70% !important;
  width: unset !important;
  height: auto !important;
  margin: auto;
  top: 50% !important;
  -webkit-transform: translateY(-50%);
  transform: translateY(-50%);
  text-align: center;
  overflow: hidden;
  -o-object-fit: contain;
  object-fit: contain;
  display: block;
}

.image-overlay {
  position: absolute;
  top: 0;
  bottom: 0;
  height: 100%;
  width: 100%;
  opacity: 0;
  -webkit-transition: 0.4s ease;
  transition: 0.4s ease;
  background-color: $color-primary;
  border-radius: 8px;
  z-index: 100;

  &:hover {
    opacity: 0.7;
  }

  @include for-tablet-portrait-down {
    background-color: transparent;
    opacity: 1;
  }
}

.image-overlay-text {
  color: $color-white;
  padding-top: 40px;
  padding-left: 40px;
  max-width: 60%;
  line-height: 1;

  @include for-tablet-portrait-down {
    background-color: $color-primary;
    color: $color-white;
    padding: 8px 0px 1px 16px;
    border-radius: 0 0 16px;
  }
}

.project-company {
  position: relative;
  font-size: 12pt;
  font-weight: 500;
  bottom: 3px;
}

.project-title {
  font-size: 20pt;
  font-weight: 900;
}
