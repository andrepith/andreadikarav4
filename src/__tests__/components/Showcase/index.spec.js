import renderer from "react-test-renderer";
import Showcase from "../../../components/Showcase";

const portofolio = [
  {
    url: "https://spacestock.com/",
    image:
      "https://res.cloudinary.com/dqajgwdwt/image/upload/v1615825499/spacestock-web.png",
    alt: "spacestock-web",
    name: "Spacestock",
    type: "Website",
  },
];

it("Renders correctly", () => {
  const tree = renderer.create(<Showcase portofolio={portofolio} />).toJSON();
  expect(tree).toMatchSnapshot();
});
