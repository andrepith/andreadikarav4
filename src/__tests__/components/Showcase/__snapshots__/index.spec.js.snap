// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<section
  className="justify-content-start showcase"
  id="project"
>
  <div
    className="container"
  >
    <h2
      className="section-title text-center"
    >
      Project.
    </h2>
    <div
      className="row"
    >
      <div
        className="col-lg-6 col-sm-12 mt-3 showcase-cards"
      >
        <a
          className="showcase-card"
          href="https://spacestock.com/"
          target="__blank"
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "left": 0,
                "margin": 0,
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt="spacestock-web"
              className="showcase-image image"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                alt="spacestock-web"
                className="showcase-image image"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdqajgwdwt%2Fimage%2Fupload%2Fv1615825499%2Fspacestock-web.png&w=3840&q=75"
                srcSet="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdqajgwdwt%2Fimage%2Fupload%2Fv1615825499%2Fspacestock-web.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdqajgwdwt%2Fimage%2Fupload%2Fv1615825499%2Fspacestock-web.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdqajgwdwt%2Fimage%2Fupload%2Fv1615825499%2Fspacestock-web.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdqajgwdwt%2Fimage%2Fupload%2Fv1615825499%2Fspacestock-web.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdqajgwdwt%2Fimage%2Fupload%2Fv1615825499%2Fspacestock-web.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdqajgwdwt%2Fimage%2Fupload%2Fv1615825499%2Fspacestock-web.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdqajgwdwt%2Fimage%2Fupload%2Fv1615825499%2Fspacestock-web.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdqajgwdwt%2Fimage%2Fupload%2Fv1615825499%2Fspacestock-web.png&w=3840&q=75 3840w"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
          <div
            className="image-overlay"
          >
            <div
              className="image-overlay-text"
            >
              <p
                className="project-company mb-0"
              >
                Spacestock
              </p>
              <p
                className="project-title"
              >
                Website
              </p>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</section>
`;
